phpseclib/Net/SSH2.php

```diff
@@ -2132,9 +2144,11 @@ function _keyboard_interactive_process()
      */
     function _ssh_agent_login($username, $agent)
     {
+        $this->agent = $agent;
         $keys = $agent->requestIdentities();
         foreach ($keys as $key) {
             if ($this->_privatekey_login($username, $key)) {
+                $this->agent->_on_login_success($this);
```

Comment:
It only does this if you request it (i.e.) from startSSHForwarding - that would be expected behaviour wouldn't it? As for private/protected. Since the language itself doesn't complain, I'm using this like a friend function. I basically don't want this method to be used anywhere else. There may be a more PHP idiomatic way to to do this?.